xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -48,92 +46,30 @@
  */
 abstract class XdsClient {
 
-  static final class LdsUpdate implements ResourceUpdate {
-    // Total number of nanoseconds to keep alive an HTTP request/response stream.
-    final long httpMaxStreamDurationNano;
-    // The name of the route configuration to be used for RDS resource discovery.
-    @Nullable
-    final String rdsName;
-    // The list virtual hosts that make up the route table.
-    @Nullable
-    final List<VirtualHost> virtualHosts;
-    // Filter instance names. Null if HttpFilter support is not enabled.
-    @Nullable final List<NamedFilterConfig> filterChain;
-    // Server side Listener.
+  @AutoValue
+  abstract static class LdsUpdate implements ResourceUpdate {
+    // Http level api listener configuration.
     @Nullable
-    final Listener listener;
-
-    LdsUpdate(
-        long httpMaxStreamDurationNano, String rdsName,
-        @Nullable List<NamedFilterConfig> filterChain) {
-      this(httpMaxStreamDurationNano, rdsName, null, filterChain);
-    }
+    abstract HttpConnectionManager httpConnectionManager();
```

Comment:
On the server side there is no such thing as one httpConnectionManager in the LDS because you can have more than one httpConnectionManager due to the filterChain. What's the plan for properly supporting the server side semantics?

Reply:
See `EnvoyServerProtoData.FilterChain`: One `FilterChain` has one `HttpConnectionManager`. The server side will only use `LdsUpdate.listener` field, the rests are for client side.

Reply:
Now `LdsUpdate` is an oneof `HttpConnectionManager` (for client-side) or `Listener` (for server-side).